### PR TITLE
common/hooks: speed up generate-runtime-deps.

### PR DESCRIPTION
**Pull request by ericonr**
Instead of using file(1) to check for ELF files, just read bytes
directly from the file and check if they are the ELF magic bytes.

Should probably be factored out into a common function that can be used
in other places.

#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [ ] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

File: common/hooks/pre-pkg/04-generate-runtime-deps.sh [hunk 1]
```diff
@@ -72,18 +72,17 @@ hook() {
...
-                for nlib in $($OBJDUMP -p "$f"|grep NEEDED|awk '{print $2}'); do
-                    [ -z "$verify_deps" ] && verify_deps="$nlib" && continue
-                    found=0
-                    for j in ${verify_deps}; do
-                        [[ $j == $nlib ]] && found=1 && break
-                    done
-                    [[ $found -eq 0 ]] && verify_deps="$verify_deps $nlib"
+        read -n4 elfmagic < "$f"
+        if [ "$elfmagic" = $'\177ELF' ]; then
+            for nlib in $($OBJDUMP -p "$f"|grep NEEDED|awk '{print $2}'); do
```
Comment (sgn) [inline common/hooks/pre-pkg/04-generate-runtime-deps.sh:77]: Maybe replace `grep NEEDED | awk '{print $2}'` with `awk '/NEEDED/{print $2}'` to save some `fork(2)`, `execve(2)`, `pipe(2)` and `dup2`

File: common/hooks/pre-pkg/04-generate-runtime-deps.sh [hunk 2]
```diff
@@ -72,18 +72,17 @@ hook() {
...
+                [ -z "$verify_deps" ] && verify_deps="$nlib" && continue
+                found=0
+                for j in ${verify_deps}; do
+                    [[ $j == $nlib ]] && found=1 && break
                 done
-                ;;
-        esac
+                [[ $found -eq 0 ]] && verify_deps="$verify_deps $nlib"
```
Comment (Piraty) [inline common/hooks/pre-pkg/04-generate-runtime-deps.sh:83]: `[`

Comment (ericonr) [reply common/hooks/pre-pkg/04-generate-runtime-deps.sh:83]: We had been using that construct for a long while now, why change?

Comment (Piraty) [reply common/hooks/pre-pkg/04-generate-runtime-deps.sh:83]: i prefer to not use `[[`  if that bashism can be avoided, but consistency is a valid point

File: common/hooks/pre-pkg/04-generate-runtime-deps.sh (hunk 1 above)
Comment (ericonr) [reply common/hooks/pre-pkg/04-generate-runtime-deps.sh]: Done and seems to be working!

**Review by ahesford — APPROVED** [review]

**Pull request merged**